go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,982 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+				// if ovnkube-master is restarted, it will gets all the add events with completed pods
+				if oc.isObjectInTerminalState(objectsToRetry.oType, obj) {
+					oc.processObjectInTerminalState(objectsToRetry, obj, key, resourceEventAdd)
+					return
+				}
+
+				// If there is a delete entry with the same key, we got an add event for an object
+				// with the same name as a previous object that failed deletion.
+				// Destroy the old object before we add the new one.
+				if retryEntry := objectsToRetry.getObjRetryEntry(key); retryEntry != nil && retryEntry.oldObj != nil {
+					klog.Infof("Detected stale object during new object"+
+						" add of type %v with the same key: %s",
+						objectsToRetry.oType, key)
+					internalCacheEntry := oc.getInternalCacheEntry(objectsToRetry.oType, obj)
+					if err := oc.deleteResource(objectsToRetry, obj, internalCacheEntry); err != nil {
+						klog.Errorf("Failed to delete old object %s of type %v,"+
+							" during add event: %v", key, objectsToRetry.oType, err)
+						oc.recordErrorEvent(objectsToRetry.oType, obj, err)
+						objectsToRetry.unSkipRetryObj(key)
+						return
+					}
+					objectsToRetry.removeDeleteFromRetryObj(key)
+				}
+				start := time.Now()
+				if err := oc.addResource(objectsToRetry, obj, false); err != nil {
+					klog.Errorf("Failed to create %v object %s, error: %v",
+						objectsToRetry.oType, key, err)
+					oc.recordErrorEvent(objectsToRetry.oType, obj, err)
+					objectsToRetry.unSkipRetryObj(key)
+					return
+				}
+				klog.Infof("Creating %v %s took: %v", objectsToRetry.oType, key, time.Since(start))
+				objectsToRetry.deleteRetryObj(key, true)
+			},
+
+			UpdateFunc: func(old, newer interface{}) {
+				// skip the whole update if old and newer are equal
+				areEqual, err := areResourcesEqual(objectsToRetry.oType, old, newer)
+				if err != nil {
+					klog.Errorf("Could not compare old and newer resource objects of type %v: %v",
+						objectsToRetry.oType, err)
+					return
+				}
+				klog.V(5).Infof("Update event received for resource %s, old object is equal to new: %v",
+					objectsToRetry.oType, areEqual)
+				if areEqual {
+					return
+				}
+
+				// get the object keys for newer and old (expected to be the same)
+				newKey, err := getResourceKey(objectsToRetry.oType, newer)
+				if err != nil {
+					klog.Errorf("Update of resource %v failed when looking up key of new obj: %v",
+						objectsToRetry.oType, err)
+					return
+				}
+				oldKey, err := getResourceKey(objectsToRetry.oType, old)
+				if err != nil {
+					klog.Errorf("Update of resource %v failed  when looking up key of old obj: %v",
+						objectsToRetry.oType, err)
+					return
+				}
+
+				// skip the whole update if the new object doesn't exist anymore in the API server
+				newer, err = oc.getResourceFromInformerCache(objectsToRetry.oType, newKey)
+				if err != nil && kerrors.IsNotFound(err) {
```

Comment:
this doesn't cover any error other than IsNotFound. So "newer" could be reassigned to nil. I think you should just do:
```
if err != nil {
      klog.Warningf("Unable to get %v %s from informer cache (perhaps it was already deleted?), skipping update: %v", objectsToRetry.oType, newKey, err)
      return

}
```